### PR TITLE
deprecated static init methods and made initWithBuilder public

### PR DESCRIPTION
**Pull request by msohailhussain**
The pull requests deprecates static "init:" methods on 8 Optimizely classes,
exposing superior "initWithBuilder:" instance methods on these classes in
their place.  These play better with Swift 4.2 .

The pull request is mean to address issue
https://github.com/optimizely/objective-c-sdk/issues/262
reported on GITHUB.COM .

### CODE REVIEW

File: OptimizelySDKCore/OptimizelySDKCore/OPTLYProjectConfig.h [hunk 1]
```diff
@@ -77,7 +77,15 @@ NS_ASSUME_NONNULL_END
 /**
  * Initialize the Project Config from a builder block.
  */
-+ (nullable instancetype)init:(nonnull OPTLYProjectConfigBuilderBlock)builderBlock;
++ (nullable instancetype)init:(nonnull OPTLYProjectConfigBuilderBlock)builderBlock
```
Comment (mikeproeng37) [inline OptimizelySDKCore/OptimizelySDKCore/OPTLYProjectConfig.h:80]: Please update the license headers in all the files touched.

File: OptimizelySDKCore/OptimizelySDKCoreTests/OptimizelySDKCoreiOSTests-Bridging-Header.h [hunk 2]
```diff
@@ -0,0 +1,5 @@
+//
```
Comment (mikeproeng37) [inline OptimizelySDKCore/OptimizelySDKCoreTests/OptimizelySDKCoreiOSTests-Bridging-Header.h]: Please add a license header for this file

File: OptimizelySDKDatafileManager/OptimizelySDKDatafileManagerTests/OptimizelySDKDatafileManageriOSTests-Bridging-Header.h [hunk 3]
```diff
@@ -0,0 +1,4 @@
+//
```
Comment (mikeproeng37) [inline OptimizelySDKDatafileManager/OptimizelySDKDatafileManagerTests/OptimizelySDKDatafileManageriOSTests-Bridging-Header.h]: License header

**Review by mikeproeng37 — CHANGES_REQUESTED** [review]

**Review by mikeproeng37 — APPROVED** [review]

**Review by thomaszurkan-optimizely — APPROVED** [review]
LGTM

**Pull request merged**